InteractiveHtmlBom/ecad/genericjson.py

```diff
@@ -79,22 +123,29 @@ def _parse(self):
                               .format(self.file_name))
             return None, None
 
-        pcbdata = pcb['pcbdata']
-        components = [Component(**c) for c in pcb['components']]
+        self.pcbdata = pcb['pcbdata']
+        self.components = [Component(**c) for c in pcb['components']]
```

Comment:
Although python allows creating new object fields anywhere it's considered good practice to always declare all fields in constructor, some linters will even enforce it.

But there is no reason for this here, you can just pass pcbdata into functions.